bindings/pyroot/pythonizations/python/ROOT/_pythonization/_uhi.py

```diff
@@ -1,23 +1,296 @@
+    dim = self.GetDimension()
+    if isinstance(index, dict):
+        expanded_index = [slice(None)] * dim
+        for axis, value in index.items():
+            expanded_index[axis] = value
+        index = tuple(expanded_index)
+
+    if not isinstance(index, tuple):
+        index = (index,)
+
+    if index.count(...) > 1:
+        raise IndexError("Only one ellipsis is allowed in the index.")
+
+    if any(idx is ... for idx in index):
+        expanded_index = []
+        for idx in index:
+            if idx is ...:
+                break
+            expanded_index.append(idx)
+        # fill remaining dimensions with `slice(None)`
+        expanded_index.extend([slice(None)] * (dim - len(expanded_index)))
+        index = tuple(expanded_index)
+
+    if len(index) != dim:
+        raise IndexError(f"Expected {dim} indices, got {len(index)}")
+
+    return [_compute_uhi_index(self, idx, axis) for axis, idx in enumerate(index)]
+
+
+def _setbin(self, index, value):
+    """Set the bin content for a specific bin index"""
+    self.SetBinContent(index, value)
+
+
+def _resolve_slice_indices(self, index, axis):
+    """Resolve slice start and stop indices for a given axis"""
+    start, stop = index.start, index.stop
+    start = start(self, axis) if callable(start) else start or (_underflow(self, axis))
+    stop = stop(self, axis) if callable(stop) else stop or (_overflow(self, axis) + 1)
+    if start < _underflow(self, axis) or stop > (_overflow(self, axis) + 1) or start > stop:
+        raise IndexError(f"Slice indices {start, stop} out of range for axis {axis}")
+    return start, stop
+
+
+def _apply_actions(hist, actions):
+    """Apply rebinning or summing actions to the histogram, returns a new histogram"""
+    if not actions or all(a is None for a in actions):
+        return hist
+
+    if any(isinstance(a, _rebin) for a in actions):
+        rebins = [a.ngroup if isinstance(a, _rebin) else 1 for a in actions]
+        hist = _rebin(rebins)(hist)
+
+    if any(a is _sum for a in actions):
+        sum_axes = tuple(i for i, a in enumerate(actions) if a is _sum)
+        return _sum()(hist, sum_axes)
+
+    if any(a is not None and not isinstance(a, (_rebin, _sum)) for a in actions):
+        raise ValueError("Unsupported action detected in actions")
+
+    return hist
+
+
+def _get_processed_slices(self, index):
+    """Process slices and extract actions for each axis"""
+    processed_slices, actions = [], [None] * self.GetDimension()
+    for i, idx in enumerate(index):
+        if isinstance(idx, slice):
+            processed_slices.append(range(idx.start, idx.stop))
+            actions[i] = idx.step
+        else:
+            processed_slices.append([idx])
+
+    return processed_slices, actions
+
+
+def _get_slice_indices(slices):
+    """Generate all combinations of slice indices"""
+    import numpy as np
+
+    return np.array(np.meshgrid(*slices)).T.reshape(-1, len(slices))
```

Comment:
I think a comment showing a simple example with a few slices of what this line is doing would help here